scrapy/core/downloader/handlers/aiohttp.py

```diff
@@ -0,0 +1,52 @@
+"""Download handlers for http and https schemes"""
+import logging
+import asyncio
+
+import aiohttp
+from twisted.internet import defer
+
+from scrapy.http import Headers
+from scrapy.responsetypes import responsetypes
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPDownloadHandler(object):
+
+    def __init__(self, settings):
+        self.settings = settings
+
+    def download_request(self, request, spider):
+        return _force_deferred(self._download_request(request, spider))
+
+    def _download_request(self, request, spider):
```

Comment:
thanks for reviewing @asvetlov, the code is intended to be a proof of concept to clear the path to python3 support.

I was wondering how it works without decoration, my best guess is that wrapping generator with `asyncio.async()` is doing that. In any case I agree it should be properly decorated.

The previous version (previous commit) used Future api and was more Deferred centric.